### PR TITLE
Upgrade to Realm Core v13.26.0

### LINKED ISSUE #6403: Upgrade to Realm Core 13.26.0

### PR DESCRIPTION
**Pull request by kneth**
## What, How & Why?

This is an upgrade of Realm Core. I have worked around the breaking change related to the implementation of `new Realm.App()` . We might want to discuss if we would like to expose app caching at a later time.

The new base URL functionality isn't exposed yet. It will be part of roaming project (yet to be decided).

Moreover, `spec.yml` needed a [small update](https://github.com/realm/realm-core/pull/7284) so the upgrade isn't as clean as I would like it.

This closes #6403 

## ☑️ ToDos

* [x] 📝 Changelog entry
* [ ] 📝 `Compatibility` label is updated or copied from previous entry
* [ ] 📝 Update `COMPATIBILITY.md`
* [ ] 🚦 Tests
* [ ] 📦 Updated internal package version in consuming `package.json`s (if updating internal packages)
* [ ] 📱 Check the React Native/other sample apps work if necessary
* [ ] 💥 `Breaking` label has been applied or is not necessary

### CODE REVIEW

**Comment by kraenhansen** [comment]
The changes seem fine, but I suspect the `@realm/react` unit tests caused a crash.

File: integration-tests/tests/src/hooks/import-app-before.ts [hunk 1]
```diff
@@ -48,6 +48,7 @@ export function importAppBefore(config: AppConfig | { config: AppConfig }, sdkCo
     } else {
       const { appId, baseUrl } = await importApp(config);
       this.app = new Realm.App({ id: appId, baseUrl, ...sdkConfig });
+      this.baseUrl = baseUrl;
```
Comment (kraenhansen) [inline integration-tests/tests/src/hooks/import-app-before.ts:51]: I would probably just import this directly in the test, but if you want to store this in the context, it should probably be added to the `AppContext` type as well.

File: integration-tests/tests/src/tests/sync/user.ts [hunk 2]
```diff
@@ -677,9 +678,10 @@ describe.skipIf(environment.missingServer, "User", () => {
       const credentials = Realm.Credentials.anonymous();
       const user = await this.app.logIn(credentials);
       const appId = this.app.id;
+
       //@ts-expect-error Wanting to prove that a completely new app instance will still return the current logged in user
       delete this.app;
-      this.app = new Realm.App({ id: appId, baseUrl: this.baseUrl });
+      this.app = new Realm.App({ id: appId, baseUrl: baseUrl });
```
Comment (kraenhansen) [inline integration-tests/tests/src/tests/sync/user.ts:684]:
```suggestion
      this.app = new Realm.App({ id: appId, baseUrl });
```

**Comment by kraenhansen** [comment]
Have you had a chance to look into the crash from the `@realm/react` tests? It looks pretty deterministic.

File: packages/realm/bindgen/js_opt_in_spec.yml
Comment (elle-j) [inline packages/realm/bindgen/js_opt_in_spec.yml:1]: I read the PR description, but from the Core [spec.yml changes](https://github.com/realm/realm-core/compare/7227d6a447821c28895daa099b6c7cd4c99d461b...dcab5907983e4441de501c97ac33962b6fc85ca1#diff-9d9d60c03d9a60ce8f2a49a0bd07324beae03010bba8b280bb697fb38b0be09f) it does look like the base URL related changes are also there.

**Review by elle-j — COMMENTED** [review]
Nice upgrade! Just have a comment about the base URL changes that would be good to address 🙂

File: packages/realm/bindgen/js_opt_in_spec.yml
Comment (kneth) [reply packages/realm/bindgen/js_opt_in_spec.yml:1]: My impression is that it should be done later but I might be wrong.

**Comment by kneth** [comment]
> Have you had a chance to look into the crash from the @realm/react tests?

It is `someurl` in https://github.com/realm/realm-js/blob/main/packages/realm-react/src/__tests__/AppProvider.test.tsx#L30 - it is a malformed URL, and modifying it to `http://someurl` makes the test pass.

File: packages/realm/bindgen/js_opt_in_spec.yml
Comment (elle-j) [reply packages/realm/bindgen/js_opt_in_spec.yml:1]: Ah I see now that `base_url` was removed from the `App` and not `AppConfig` in the spec 👍

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,18 +7,18 @@
 * Added an optional `fetch` parameter to the `AppConfiguration`. Use this to specify a custom implementation of the `fetch` function used by the app to perform network requests.
 
 ### Fixed
-* <How to hit and notice issue? what was the impact?> ([#????](https://github.com/realm/realm-js/issues/????), since v?.?.?)
-* None
+* Handle `EOPNOTSUPP` when using `posix_fallocate()` and fallback to manually consume space. This should enable android users to open a Realm on restrictive file systems. ([#6349](https://github.com/realm/realm-js/issues/6349), since v12.4.0)
```
Comment (elle-j) [inline CHANGELOG.md:10]: In the issue is said that it happened after upgrading to `v12.2.0`.

**Review by elle-j — APPROVED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (kneth) [reply CHANGELOG.md:10]: You have a point, and the core change log actually says "... also an issue in some platforms before this".

**Comment by kraenhansen** [comment]
I've created https://github.com/realm/realm-js/issues/6441 to track fixing the crash 👍

**Review by kraenhansen — APPROVED** [review]

**Pull request merged**